lib/matplotlib/axes/_axes.py

```diff
@@ -2419,24 +2421,37 @@ def stem(self, *args, **kwargs):
             basefmt = kwargs.pop('basefmt', args[2])
         except IndexError:
             basefmt = kwargs.pop('basefmt', 'r-')
+        try:
+            orientation = kwargs.pop('orientation', args[3])
+        except IndexError:
+            orientation = kwargs.pop('orientation', 'vertical')
 
         bottom = kwargs.pop('bottom', None)
         label = kwargs.pop('label', None)
-
-        markerline, = self.plot(x, y, markerfmt, label="_nolegend_")
+        if (orientation == "vertical"):
```

Comment:
@QuLogic Hi I think it need both vertical and horizontal check there or else it will plot marker lines twice, unless you meant a else statement instead of elif like:
if (orientation == 'horizontal'):
    ...
else:
  ...